cylc/flow/config.py

```diff
@@ -844,12 +845,19 @@ def _check_circular(self):
         """Check for circular dependence in graph."""
         start_point_string = (
             self.cfg['visualization']['initial cycle point'])
+        raw_graph = self.get_graph_raw(start_point_string,
+                                       stop_point_string=None)
+        if (len(raw_graph) > self.CHECK_CIRCULAR_LIMIT and not (
```

Comment:
Can use `len(self.taskdesf)` to measure the number of tasks rather than graph edges (`get_graph_raw` is slow).